content/blogposts/2023/guest-post-arafat-khan-php-python.md

```diff
@@ -0,0 +1,256 @@
+---
+title: "Climbing Mount Everest in Flip-flops: My PHP journey as a Python dev"
+authors:
+  - name: Arafat Khan
+    url: https://twitter.com/arafatkatze
+  
+publishDate: 2023-11-21T10:00-01:00
+description: "When tasked with creating a printing API in PHP despite having zero experience, Arafat Khan leveraged an AI coding assistant named Cody to transform what felt like an Everest climb into a swift 3 day project that resulted in a fully functioning API."
+tags: [blog]
+slug: "guest-post-arafat-khan-php-python"
+published: true
+heroImage: https://storage.googleapis.com/sourcegraph-assets/blog/guest-post-arafat-khan-php-python-og.png
+socialImage: https://storage.googleapis.com/sourcegraph-assets/blog/guest-post-arafat-khan-php-python-og.png
+---
+
+<Alert type="secondary">This is a guest blog post from our community. <a href="https://discord.com/servers/sourcegraph-969688426372825169" target="_blank">Join our Discord</a> to meet the author and connect with other developers passionate about Code Search and AI!</Alert>
+
+_Ever thought about climbing Mount Everest in flip-flops?_ That’s how I felt diving into _PHP_ for the first time.
+
+A few weeks ago, a professional endeavor landed on my desk creating a printing API. The goal was to build a library in PHP that lets developers easily play with commands to scan and print documents programmatically. With a functional Python API version already in my arsenal, I was initially optimistic about the task. Translating the code to another language, PHP in this case, seemed like a straight path. However, the catch was my complete lack of familiarity with PHP, making the experience seem like a herculean effort at first glance. To add to the challenge, I imposed a time constraint on myself so now I had to learn a new language and build a full-blown project that I could share with the world.
```

Comment:
A few weeks ago, a professional endeavor landed on my desk creating a printing API. => A few weeks ago, a professional endeavor landed on my desk: creating a printing API.

With a functional Python API version already in my arsenal => With an already functional Python API version in my arsenal

To add to the challenge, I imposed a time constraint on myself so now I had to learn a new language and build a full-blown project that I could share with the world. => To add to the challenge, I imposed a time constraint on myself. So now I had to learn a new language and build a full-blown project that I could share with the world.